src/Servers/HttpSys/src/AsyncAcceptContext.cs

```diff
@@ -50,59 +56,53 @@ internal ValueTask<RequestContext> AcceptAsync()
                 return ValueTask.FromException<RequestContext>(new HttpSysException((int)statusCode));
             }
 
-            return new ValueTask<RequestContext>(this, _mrvts.Version);
+            if (statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_IO_PENDING)
+            {
+                return new ValueTask<RequestContext>(this, _mrvts.Version);
+            }
+
+            var requestContext = _requestContext;
+            _requestContext = null;
+            return ValueTask.FromResult(requestContext);
```

Comment:
Nit: This would read better as a switch. It took me longer than I want to admit to realize the above is handling every possible non-ERROR_SUCCESS statusCode prior to this.